lib/web_ui/lib/ui_web/src/ui_web/initialization.dart

```diff
@@ -55,3 +55,68 @@ Future<void> bootstrapEngine({
     loader.didCreateEngineInitializer(bootstrap.prepareEngineInitializer());
   }
 }
+
+/// The signature of the [detectCrawler] function.
+typedef CrawlerDetector = bool Function();
+
+/// Determines whether the current user agent is a web crawler, such as a search
+/// engine.
+///
+/// The web engine calls this method once during initialization, and then it no
+/// longer calls this method. Overriding this function will have no effect after
+/// it has been called.
+///
+/// Many crawlers ignore ARIA attributes as a source of app semantics. Therefore,
+/// if a crawler is detected, the web engine outputs semantic labels as text in
+/// a DOM `<span>` element, which crawlers know how to read.
+///
+/// This function can be overriden by the app, by setting it to a different
+/// function. This can be useful when the app needs to support a crawler that's
+/// not known to the Flutter SDK.
+///
+/// The default implementation is [defaultDetectCrawler].
+CrawlerDetector detectCrawler = defaultDetectCrawler;
+
+// The list of known crawler user agent patterns. It should capture all of the
+// most important crawlers, and so it may change over time. However, it is not
+// meant to be comprehensive. Instead, the user can override `detectCrawler`
+// with a custom implementation. Developers can also share packages on pub.dev
+// that have sophisticated ways of detecting crawlers.
+final List<RegExp> _crawlerRegexes = <RegExp>[
+  // Covers the search engine and various specialized crawlers: https://developers.google.com/search/docs/crawling-indexing/overview-google-crawlers
+  RegExp(r'googlebot'),
+  RegExp(r'\-google'),
```

Comment:
This a good start though it doesn't capture all the user agents that Google lists in that url (e.g. `Google-Extended`, `GoogleOther`).

This is fine for now but let's file an issue maybe to improve the regexes?

Reply:
I'm actually starting to wonder if the engine should contain any crawler patterns at all. Maybe these belong in a package?